### DISCUSSION TITLE
Q: Informations (yet?) missing from the Wiki?

### DISCUSSION THREAD

**Discussion by esksp** [endless-sky/endless-sky#10503] [Ideas]
Hi, I'm trying to make a human campaign side branch, something a little less standard (no "go there, do this"), but of course I hit a snag.
Your Wiki howtos on content creation are good, but _very_ succinct (options are just mentioned, not explained), and since the newer features aren't used in the game version I have (10.8), there is no way to know what the right syntax might be. (And no, I'm not a developer, sifting through the game's source code is not an option!...)
Random examples of questions: System links can apparently have attributes, but which ones and how does one use those (can I make a link "shrouded" for instance?)? Or, how does on make landing on a planet only accessible to bearers of a specific license (there are some "required:" type attributes in the map-systems.txt, but nothing conclusive)?
And so many more.

**Is there a place I can find more in-depth information about content creation?** Steam seems mostly a player forum, and this place here is more for game engine development I guess. As for Discord, I know it's very hip, but I'm not a huge fan of instant messaging (I don't want to chat, I just want some information to get work done). Besides they ask for my phone number and there is no way they'll ever get it, I already get enough "exciting offers"...
TIA

**Comment by esksp** [comment]
Okay, after having been suspended 2 weeks by Github for registering (!?!), I'm back... -rolling eyes-
By now I have figured out some of the above questions, managed to create systems, set up governments, create fleets and--most surprisingly--eventually make everything behave the way I wanted!
Yet, after 2 weeks of tweaking and playing there are still some minor things I think could easily be improved. Please bear with me:

**Game itself:**
1. Could you please add **some more indicators in the HUD**? A single red letter would be enough. I'm missing indicators for "auto-target" (is it on or off?), flotsam collection (same), are your fighters deployed, which AI configuration are your escorts in, and such. Those informations might appear in the scrolling log, but it moves too fast to be of any use in a situation.
2. In the same spirit: Could we have **a different highlight for your flagship** than the other ships in your fleet? If you have more than one of a model and they fly in "grouped" (the only usable) formation, you never know which ship you're steering...
3. While we're talking about escorts: Could we _please_ get a **"stay out of this fight!" escort command**? Your escorts are _always_ acting like a bunch of rabid kamikazes: Sometimes you just want (or need) to run, but your escorts always want to slug it out, no matter the odds.
4. Escorts again: Could we have a **special marker in the radar telling you where your fighters are**? Unless on "Group", they tend to go off to explore the world, and are apparently too small to really show on the radar. Even worse when they are disabled and drifting (which they often are).
5. Outfitter: The way to **only show the outfits you carry in your cargo** is a little convoluted, it also took me a while to figure out, yet selling captured outfits is something you do (I do...) pretty often. Could we please get a more immediate way to only show outfits stored in cargo, like for instance a fourth checkbox on the bottom left ("show outfits stored in your cargo holds")?
6. A small annoyance: Most of the (otherwise great!) campaign is in pure text mode, somewhere on the left border of the screen. Since you spend most of your time squinting at that panel, I'd suggest to **make the text bigger and put it in the middle**. It's not some accessory, it is where you spend a huge part of your game time... Don't get me wrong, I don't mind reading, I just suggest to give it the importance it has.

**Content creation:**
1. As I said above, a little **more details on syntax**. Not everybody is a programmer... And some more information about what some options actually do. The problem is that usually you'd check how the game's developer have used said feature, but in this case most features are not used anywhere, it's up to you to guess what they do and how they work.
2.  The star map ("**endless-sky-editor**") program is too well hidden!... It took me day to find a way from https://github.com/endless-sky/endless-sky-editor to https://github.com/endless-sky/endless-sky-editor/releases and find an usable download link... I'd suggest to add that link to the "MapData" Wiki page.

So, guess I'll get locked out again for a week or two, so I'll just add, **great game, really outstanding work, thanks a lot guys!**

**Reply by TheGiraffe3** [reply]
I'm sorry to hear that you got suspended. It can be frustrating (saying this from personal experience).
Also, discussions you make get hidden during suspension, so I don't think anybody realized this existed.

2. That's an option already.
3. You can (in beta) press Y. That'll tell them to hold fire.

<br>
1. I'm not a coder (at all), but it's not hard to understand. You can read the pull requests that added what you want to know for usage examples.<br>
2. Sorry for telling you this only now, but there is a better editor. https://github.com/quyykk/editor

**Reply by esksp** [reply]
Thank you for your reply!

About 2: You can indeed highlight your flagship and escorts, but (at least for me, Linux, v.0.10.8) they both use the exact same green highlight, thus the confusion... Could be have a (for instance) _white_ highlight for the flagship?

About 3: In beta, huh? I'll wait till it goes mainstream, I'm already messing around enough to want to be sure I am the culprit if something goes south... :-D

"Pull requests"? I have only a vague idea what that is... Sorry. I'm using computers from times before the IBM PC was invented, I'm fairly proficient, but I'm no developer and all this developer talk is above my pay grade...

https://github.com/quyykk/editor: Quote: "This project has been discontinued and has been replaced by https://github.com/quyykk/plugin-editor." Thanks, it sounds great, but my point still stands: There should be a mention in the Wiki...

**Reply by TheGiraffe3** [reply]
You should be able to highlight your flagship only.
By beta I mean 0.10.9.

A "pull request" is a request for code to be pulled into the main repository. So for example, #6610 was the "pull request" that when merged added the Gegno into the game.


What is the specific feature in 0.10.9 that isn't well documented?


There is a mention in the wiki of the old editor, and I have a pull request that will point it to the new editor.

**Reply by esksp** [reply]
Highlighting: But what if I want (need) to **highlight _both_ my flagship and the escorts**? I need to keep an eye on which ships I'm supposed to care for (not only my own, but also and mostly ships during escort missions).

Thanks for the "pull request" explanation and the Wiki update!

As for "not ideally" documented features, there are a few indeed (BTW mine is v.0.10.**8** actually).
Many game mechanisms are not very well explained, take "**mining**" for instance. You immediately guess you need to break the asteroids apart with a mining gun, but how do you collect the pieces? The manual is quite vague about that. Flying over them doesn't do anything, and hitting the "collect flotsam" hotkey doesn't seem to change that (but then I don't even know if that feature is now on or off, a HUD indicator would really be nice, just a "$" sign or some such).
On the other hand I once pushed that "flotsam" hotkey by error, and when I wanted to collect some expensive piece of kit from a neutralized spaceship, my cargo holds were already full of worthless flotsam!... Imagine barely fighting a strong and rare enemy, managing to immobilize him, only to discover that somehow your cargo is now full with worthless "clothes" and "food"!... (You're probably going to tell me there is a way to jettison unwanted cargo, but I haven't found it yet.)
I scoured the forums (here and Steam) to find some really indispensable pieces of knowledge, like for instance how to handle the outfitter GUI (items on sale, installed, cargo, storage).

**Reply by warp-core** [reply]
> 1. Could you please add **some more indicators in the HUD**? A single red letter would be enough. I'm missing indicators for "auto-target" (is it on or off?), flotsam collection (same), are your fighters deployed, which AI configuration are your escorts in, and such. Those informations might appear in the scrolling log, but it moves too fast to be of any use in a situation.

Feature requests like these would be most helpfully places in [issues](https://github.com/endless-sky/endless-sky/issues).

esksp:
> 2. In the same spirit: Could we have **a different highlight for your flagship** than the other ships in your fleet? If you have more than one of a model and they fly in "grouped" (the only usable) formation, you never know which ship you're steering...

TheGiraffe3:
> That's an option already.

esksp:
> About 2: You can indeed highlight your flagship and escorts, but (at least for me, Linux, v.0.10.8) they both use the exact same green highlight, thus the confusion... Could be have a (for instance) white highlight for the flagship?

Are you referring to the "status overlay" that shows the current health of each ship? These colours are indeed the same for the flagship and escorts by default. The colours can be customised to suit you. I'm working on a page for the wiki that better describes this, but for now, [here's a explanation I wrote earlier](https://github.com/endless-sky/endless-sky/discussions/8614#discussioncomment-5492338).
If you would like to change the defaults, you can open an issue or PR suggesting new values, 

If you are referring to the "Highlight player's flagship" option in the HUD section of the settings, this isn't applied to escorts and should only appear on the flagship. You can change the colour similarly to what is described in the above link, but the name of the colour is "flagship highlight".

> 3. While we're talking about escorts: Could we _please_ get a **"stay out of this fight!" escort command**? Your escorts are _always_ acting like a bunch of rabid kamikazes: Sometimes you just want (or need) to run, but your escorts always want to slug it out, no matter the odds.

If you are trying to leave an engagement, telling your escorts to gather around you should have them follow you. In the next release, you should also be able to tell them to hold fire while doing this, but this functionality currently has some issues that need to be resolved and is only available in the beta branch on Steam and GOG or the unstable release here.

> 1. As I said above, a little **more details on syntax**. Not everybody is a programmer... And some more information about what some options actually do. The problem is that usually you'd check how the game's developer have used said feature, but in this case most features are not used anywhere, it's up to you to guess what they do and how they work.

If there are items that you feel are not sufficiently well described in the wiki, please open an issue at https://github.com/endless-sky/endless-sky-wiki/issues with details of what is inadequate.

**Reply by esksp** [reply]
> Feature requests like these would be most helpfully places in [issues](https://github.com/endless-sky/endless-sky/issues).

Thanks.
Sorry, I'm new to your system and don't know yet my way around here.
Besides, it isn't (IMHO) not really an "issue", rather a improvement _suggestion_.
But I'll start a new thread in "issues".


> Are you referring to the "status overlay" that shows the current health of each ship? These colours are indeed the same for the flagship and escorts by default. The colours can be customised to suit you.

Indeed, because the actual "highlight" is (as far as I have seen) just a pixel-wide outline, which is invisible when zooming out, and I _always_ play totally zoomed out.
It would be nice if we could have the flagship "status overlay" in another color, so you can keep your ships apart when you have a bunch of identical ones.
I'll write this in that new thread in "issues".


> If you are referring to the "Highlight player's flagship" option in the HUD section of the settings

As I said above, this seems to only consist of a pixel-wide white outline, very difficult/impossible to see when zooming out. At least on my computer (Linux, v.0.10.8, graphics: Intel + NVIDIA GeForce® RTX 3070 Ti 8Go). I might be wrong, but after playing with the settings, I didn't find a satisfying way to keep my flagship apart. 


> If you are trying to leave an engagement, telling your escorts to gather around you should have them follow you. In the next release, you should also be able to tell them to hold fire while doing this, but this functionality currently has some issues that need to be resolved and is only available in the beta branch on Steam and GOG or the unstable release here.

Leaving engagements indeed (instead of rushing to fight), but there is also the more general issue of having different ships for different tasks: Slow, lightly armed cargo ships of mine should be on "timid", if not "coward" (to use your NPC personalities), while my warships should defend my weaker ships, instead of rushing to fight their own war somewhere on the other side of the system map...
There doesn't seem to be a "protector" personality, but it shouldn't be too difficult to implement: "_If_ ship=friend _and_ ship=weak/damaged _and_ ship=under attack, _then_ go attack his attackers".
Ideally you should allow the player to set personalities for all his non-flagship ships: Aggressive for warships, timid or pacifist for cargoes. One could overlay a personality letter over the little escort list icons on the bottom left.
I'll put that in issues.


> If there are items that you feel are not sufficiently well described in the wiki, please open an issue at https://github.com/endless-sky/endless-sky-wiki/issues with details of what is inadequate.

Thanks, will do.

**Comment by warp-core** [comment]
> System links can apparently have attributes, but which ones and how does one use those (can I make a link "shrouded" for instance?)?

I'm not aware of links between systems having attributes. Could you share where you came across this? The closest thing I can think of is "conditions" which was a feature that was removed shortly after it was added (before any release) because of severe performance impacts. I note that, upon removal, the wiki was not updated to remove the description of that functionality. I've made that change now.

> Or, how does on make landing on a planet only accessible to bearers of a specific license (there are some "required:" type attributes in the map-systems.txt, but nothing conclusive)?

Landing on a planet cannot be directly restricted to requiring a license. If the license is granted by event or mission/conversation action, I suggest adjusting the required reputation of the planet at that time to allow the player to land. If the license is purchasable, it won't be so simple.

The meaning of the "requires: <attribute>" attribute is described in https://github.com/endless-sky/endless-sky/wiki/MapData#planets, which states:

> If the "requires: <attribute>" phrase is used, then a ship must have the listed attribute in order to land on this planet.

Reading this section, I can see some potential for formatting improvements that would make it easier to find and associate the text with certain things.

> **Is there a place I can find more in-depth information about content creation?** Steam seems mostly a player forum, and this place here is more for game engine development I guess. As for Discord, I know it's very hip, but I'm not a huge fan of instant messaging (I don't want to chat, I just want some information to get work done). Besides they ask for my phone number and there is no way they'll ever get it, I already get enough "exciting offers"... TIA

Discord is by far the most populous and active place of discussion about the game that I am aware of, but if you don't want to use it, the Steam forums and this forum are both places where you can find help with content creation. I would suggest here over Steam, since you're already here. There are also the GOG forums, but that is relatively new for us, so there isn't really a whole lot of activity over there, and Reddit, but I don't know how content creation help over there usually goes.

**Reply by esksp** [reply]
> I'm not aware of links between systems having attributes. Could you share where you came across this?

I admit I don't remember where I found it, somewhere in the Wiki, but it's been weeks ago and I quickly ceased searching.
My goal was to make a branch of systems you can't see on the map and only visit if you know how to get there.
I managed that using the "shrouded" attribute, but I was looking for a way to also remove that suspicious link pointing to nowhere (from the last "normal" system), a way to make it invisible _unless_ you actually are _in_ that system.
Maybe making that links to "shrouded" systems are also "shrouded"? It would make sense (to me at least). I'll put in my "issues" suggestion list post.

> Landing on a planet cannot be directly restricted to requiring a license. If the license is granted by event or mission/conversation action, I suggest adjusting the required reputation of the planet at that time to allow the player to land. If the license is purchasable, it won't be so simple.

Thanks, that's what I discovered. Adding a license (visa?) requirement would be nice.
Will put in my "issues" suggestion list post.


> The meaning of the "requires: " attribute is described in https://github.com/endless-sky/endless-sky/wiki/MapData#planets, which states:
> 
> > If the "requires: " phrase is used, then a ship must have the listed attribute in order to land on this planet.
> 
> Reading this section, I can see some potential for formatting improvements that would make it easier to find and associate the text with certain things.

Indeed, because which attributes are possible? For instance, a license isn't a valid attribute...
A list of attributes understood would be nice.


> Discord is by far the most populous and active place of discussion about the game that I am aware of, but if you don't want to use it, the Steam forums and this forum are both places where you can find help with content creation.

This forum here is apparently the most appropriate place, except I have issues with GitHub, or rather, they have issues with me: I'm not a GitHub user, I registered here just to post what you see in this page, and have been already automatically suspended twice! Total membership time: 3 weeks, total time my account was actually active: 3 hours...

> I would suggest here over Steam, since you're already here. 

Actually I'm not, I found your game on GOG. But the GOG forum is empty (3-4 posts), so asking in-depth questions there is quite pointless.
Oh well, I'll try to stick with this forum here, if only GitHub loses that bad habit of suspending me every time I try to log in ("Successive login attempts from the same IP"! My, how utterly suspect!!!)...

**Reply by warp-core** [reply]
If you are already on GOG, you can use the forums there. You can probably expect help from at least the maintainers, even if there isn't a large community there at the moment.

**Reply by esksp** [reply]
> If you are already on GOG, you can use the forums there. You can probably expect help from at least the maintainers, even if there isn't a large community there at the moment.

I'm afraid asking in-depth question over there will be just as useful as speaking them out loud in my bathroom... :-D
Here at least I have the ear of people who know more than me.
Oh well, that's not the issue, back to the game.

**Comment by esksp** [comment]
Another question, being a newbie I'd rather ask before opening an "issue":

**Jump drives have ranges**, which show in the map as a gray circle. So far, so good.
Now imagine two identical ships, same outfits, same fuel capacity, both equipped with a jump drive: They should have the same jump range, shouldn't they.

My two identical ships did have identical ranges, except when I jumped (exploring) to the extreme east of the map (uninhabited letter+number systems), when for no reason **my escort started having a much smaller jump range**! Really smaller, about 60% of my flagship (two circles on the map). Or maybe my flagship got a bigger range? Can't tell, but they were clearly different, and this wasn't a display bug, since my escort was unable to jump further than the smaller circle, while my flagship could jump to all systems in the bigger one.

Is there some mechanism I'm not aware of, or is this a bug?

**Reply by Saugia** [reply]
This is intended; some systems in the far east have edited Jump Drive ranges so you cannot go certain ways normally. There are lore reasons in that region as to why that is the case.

**Reply by TheGiraffe3** [reply]
The question is why the escort can't make it but the player's ship can.

**Reply by esksp** [reply]
Indeed. I only started thinking about the range once it started being different for my 2 (otherwise identical) ships.
So, bug?

**Reply by TheGiraffe3** [reply]
Probably.

**Reply by esksp** [reply]
Ok, I will assemble a bug report. Unfortunately I saw this a week ago, I was waiting for my GitHub suspension to be fixed back then, and I didn't keep the savegames. But it's easy to recreate I think.
Will post a bug report tomorrow (unless GitHub suspends me again: I've been let loose for almost 12 hours, this can't last...).

Edit:
Done: Bug report #10596

**Reply by xobes** [reply]
@esksp  I have a whole page of things I'd like to see be different, knowing what my escorts *think they should be doing* is one of them.

I also want my escorts to stop shooting things -- saw there was an update that was just removed that might have been an attempt at resolving that...

I also want to be able to select not just next (always next) but previous... the list goes on.

I am grateful that we have this game and can identify ways to improve it.  However, what appears to one person to be an "improvement" doesn't seem as such to another.  I was playing 0.10.4 (cuz, Ubuntu) until the other day when I saw 0.10.10 release on github and switched.  Now I get fancy graphics for ships all over -- I'm an Escape Velocity fan here, that game made you purchase outfits to improve the crappy monochrome radar display by buying an IFF decoder, and poof, now you have colors for friendly, hostile, etc.  In that vein, I think that the automatic 3d-ship view is way overkill and I want the outlines back -- until I buy some outfit to give me awesome images of the ships.  Also, the ship colors better match, but I don't see proper color tweening there, yet.

Another thing I'd like to react to is this comment you made about "sifting through the game's source code is not an option!" -- It *is* an option.  And it may be the only option, until it isn't: If/when someone figures something out, and they are fresh and say "wow, that took way too long to figure out" -- then that person (you, me, the next person) should bear some responsibility to leave the code or wiki in a better place.

We shouldn't *have* to read the code to know how a given .txt file and all the plugin syntax is supposed to work, but unless updating the wiki becomes a mandatory step, it might be the only way.  And with the wiki being it's own thing, it's not likely to be included in every PR.  Follow that up with the fact that the people who understand how something works or is supposed to work (plugin syntax, for example) is not aware of what us newbies see, think and feel when we're starting off.

There's a lot of possibility here, I'm looking forward to learning more about how to contribute in a meaningful way and maybe I'll be able to see some of my laundry list of ideas come to fruition.

**Reply by tibetiroka** [reply]
> I think that the automatic 3d-ship view is way overkill and I want the outlines back

I think there is a preference for that. Probably something like "Ship outlines" fancy/fast.
